fplll: 5.2.1 -> 5.3.0, python.pkgs.fpylll: 0.4.1dev -> 0.5.0dev

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/fplll/versions.



meta.description for fplll is: '"Lattice algorithms using floating-point arithmetic"'.


meta.homepage for fplll is: '""


[Compare changes on GitHub](https://github.com/fplll/fplll/compare/5.2.1...5.3.0)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 5.3.0 with grep in /nix/store/00q56pdlsnpyj8mwjwf0xd2qzwl5ixwr-fplll-5.3.0
- directory tree listing: https://gist.github.com/215205c63fcbc398cd1723a6bdbb13b0

- du listing: https://gist.github.com/1f63c8aa27eba9e5b3234feeb4dfcb58


</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

18 total rebuild path(s)

6 package rebuild(s)

6 x86_64-linux rebuild(s)
4 i686-linux rebuild(s)
4 x86_64-darwin rebuild(s)
4 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
fplll
python27Packages.fpylll
python37Packages.fpylll
python38Packages.fpylll
sage
sageWithDoc

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/00q56pdlsnpyj8mwjwf0xd2qzwl5ixwr-fplll-5.3.0 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A fplll https://github.com/r-ryantm/nixpkgs/archive/72172b0f9d1e3f464757a2e1b6e02b17c4f91cff.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/00q56pdlsnpyj8mwjwf0xd2qzwl5ixwr-fplll-5.3.0
ls -la /nix/store/00q56pdlsnpyj8mwjwf0xd2qzwl5ixwr-fplll-5.3.0/bin
```


</details>
<br/>





cc @7c6f434c for testing.

7c6f434c commented:
@timokau probably has a qualified opinion

timokau commented:
I don't have any opinion other than `nix-review` should pass -- which it currently doesn't do since this breaks `fpyll`. Those two probably need to be updated together.

7c6f434c commented:
Ah right, I guess you only start forming opinion once Sage is the only thing that fails.

timokau commented:
Well that's included in nix-review passing ;)

What I meant is that I don't personally use this package, so I don't really have anything to add in addition to "it shouldn't obviously break anything".

7c6f434c commented:
Well, you might have already heard of some complaints about the new version breaking Sage. (And no other use of this package seems to be known)

timokau commented:
Good point :) There don't seem to be any [known](https://trac.sagemath.org/search?q=fplll) issues. Other than that, I usually trust the testsuite. It tends to have lots of false-positives, but rarely false-negatives.

timokau commented:
Fpyll update is not trivial, something has changed with the test setup. I've bisected it to either 327c4c1a3f86c0fc877cfe3e177a3ab3664ece5b or 9732fdb40cf1bd43ad1f60762ec0a8401743fc79.

No time to continue right now. I'll try to get back to it when I have time, but if someone else wants to finish I certainly don't mind ;)

timokau committed:
fplll: 5.2.1 -> 5.3.0, python.pkgs.fpylll: 0.4.1dev -> 0.5.0dev

The two packages need to be updated together, since fpylll 0.4.1
requires the old fplll version and fpylll 0.5.0 requires the new one.

timokau commented:
Okay the fpylll build succeeds now. There is one transient test issue (https://github.com/fplll/fpylll/issues/161) that I've skipped for now. We should give upstream a couple of days to react to that before merging.

I haven't run `nix-review` yet.

timokau commented:
`nix-review` passes.

7c6f434c commented on pkgs/development/python-modules/fpylll/default.nix:
```diff
@@ -14,13 +14,14 @@
-  version = "0.4.1dev";
+  version = "0.5.0dev";
 
+  # src = lib.cleanSource /home/timo/repos/fpylll;
```
Do you want to keep it for the next time you need to patch something?

timokau replied:
No, I just used that to bisect the issue. Didn't mean to commit it. Good catch!

jonringer commented on pkgs/development/libraries/fplll/default.nix:
```diff
@@ -3,12 +3,12 @@
-    sha256 = "015qmrd7nfaysbv1hbwiprz9g6hnww1y1z1xw8f43ysb7k1b5nbg";
+    sha256 = "0wxa4xs7as7y47h7i6prmk5r0srabdvrlkvza3j50pixir5swgvh";
   };
   nativeBuildInputs = [autoconf automake libtool gettext autoreconfHook];
```
the hook will bring in autotool binaries
```suggestion
  nativeBuildInputs = [gettext autoreconfHook];
```

jonringer reviewed (approved):
otherwise LGTM

timokau committed:
fplll: update formatting

According to my perception of the current community standards.

timokau committed:
fplll: clean up dependencies

autoreconfHook implies autotools.

timokau committed:
fplll: add changelog

timokau committed:
fplll: add timokau to the maintainers list

timokau commented:
Should be good to go now if @7c6f434c approves of the cleanup commits. If you still prefer the more concise style you used when you created the package, I can revert that change. I just thought that commits like the "clean up dependencies" one have a nicer diff with the new formatting.

The changelog is not ideal, since it links to a search. The project also has a [proper](https://github.com/fplll/fplll/blob/master/NEWS) changelog, but that has been out of date for a couple of years.

timokau commented:
The failing test is now resolved as well.

7c6f434c commented:
As I consider you more competent than myself here (because of your great Sage work), I don't have any opinions re: style.

Re: changelog — the pain, it's not _just_ search, it is JavaScript-only search… I propose https://github.com/fplll/fplll/releases as the first entry in the changelog list.

(I have read the entire diff and briefly checked the basic reasonability of the added patch)

timokau committed:
python.pkgs.fpylll: add changelog

timokau commented:
> As I consider you more competent than myself here (because of your great Sage work), I don't have any opinions re: style.

Appreciate the compliment :)

> Re: changelog — the pain, it's not _just_ search, it is JavaScript-only search… I propose https://github.com/fplll/fplll/releases as the first entry in the changelog list.

Oh neat, I didn't realise the git tags had release notes as well. I've added that to the list (although they don't seem complete -- for example the latest version doesn't have any release notes there).

> (I have read the entire diff and briefly checked the basic reasonability of the added patch)

Please have another look (just changed the changelogs) and merge if it looks good to you.

7c6f434c commented:
I wonder at what point we will just subscribe a bot to various mailing list with release announcements because it is more reliable…

timokau commented:
Unfortunately this PR broke the aarch64 build of fpylll: https://github.com/fplll/fpylll/issues/162

Though the test failures look pretty harmless, so we could probably just ignore them.
